examples/pruning/tensorflow_estimator_integration.py

```diff
@@ -24,90 +24,78 @@
+    test_ds = data['test']
+    test_ds = test_ds.map(preprocess).shuffle(10000).batch(BATCH_SIZE).take(N_TEST_BATCHES)
+    return test_ds
 
-    loss = tf.compat.v1.losses.sparse_softmax_cross_entropy(labels=labels, logits=logits)
-
-    if mode == tf.estimator.ModeKeys.TRAIN:
-        optimizer = tf.compat.v1.train.GradientDescentOptimizer(learning_rate=0.001)
-        train_op = optimizer.minimize(
-            loss=loss, global_step=tf.compat.v1.train.get_or_create_global_step())
-        return tf.estimator.EstimatorSpec(mode=mode, loss=loss, train_op=train_op)
-
-    eval_metric_ops = {
-        "accuracy": tf.compat.v1.metrics.accuracy(
-            labels=labels, predictions=predictions["classes"])
-    }
-    return tf.estimator.EstimatorSpec(mode=mode, loss=loss, eval_metric_ops=eval_metric_ops)
-
-
-def objective(trial):
-    (train_data, train_labels), (eval_data, eval_labels) = tf.keras.datasets.mnist.load_data()
 
-    train_data = train_data / np.float32(255)
-    train_labels = train_labels.astype(np.int32)
+def create_classifier(trial):
+    # We optimize the numbers of layers and their units.
 
-    eval_data = eval_data / np.float32(255)
-    eval_labels = eval_labels.astype(np.int32)
+    n_layers = trial.suggest_int('n_layers', 1, 3)
+    hidden_units = []
+    for i in range(n_layers):
+        n_units = trial.suggest_int('n_units_l{}'.format(i), 1, 128)
+        hidden_units.append(n_units)
 
     config = tf.estimator.RunConfig(
         save_summary_steps=PRUNING_INTERVAL_STEPS, save_checkpoints_steps=PRUNING_INTERVAL_STEPS)
 
     model_dir = "{}/{}".format(MODEL_DIR, trial.number)
-    mnist_classifier = tf.estimator.Estimator(
-        model_fn=lambda features, labels, mode: model_fn(trial, features, labels, mode),
+    classifier = tf.estimator.DNNClassifier(
+        feature_columns=[tf.feature_column.numeric_column('x', shape=[28*28])],
+        hidden_units=hidden_units,
         model_dir=model_dir,
+        n_classes=10,
+        optimizer=lambda: tf.keras.optimizers.Adam(learning_rate=0.01),
         config=config)
 
+    return classifier
+
+
+def objective(trial):
+    classifier = create_classifier(trial)
+
     optuna_pruning_hook = optuna.integration.TensorFlowPruningHook(
         trial=trial,
-        estimator=mnist_classifier,
+        estimator=classifier,
         metric="accuracy",
         run_every_steps=PRUNING_INTERVAL_STEPS,
     )
 
-    train_input_fn = tf.compat.v1.estimator.inputs.numpy_input_fn(
-        x={"x": train_data}, y=train_labels, batch_size=BATCH_SIZE, num_epochs=None, shuffle=True)
-
     train_spec = tf.estimator.TrainSpec(
         input_fn=train_input_fn, max_steps=TRAIN_STEPS, hooks=[optuna_pruning_hook])
 
-    eval_input_fn = tf.compat.v1.estimator.inputs.numpy_input_fn(
-        x={"x": eval_data}, y=eval_labels, num_epochs=1, shuffle=False)
-
     eval_spec = tf.estimator.EvalSpec(
         input_fn=eval_input_fn, steps=EVAL_STEPS, start_delay_secs=0, throttle_secs=0)
```

Comment:
Please remove `step=EVAL_STEPS` similarly to https://github.com/optuna/optuna/pull/924#discussion_r378791734. If `step` is set, we restrict the number of examples using two constants (i.e., `EVAL_STEPS` and `N_TEST_BATCHES`) and it seems to be a bit complicated. 
(c.f., https://www.tensorflow.org/api_docs/python/tf/estimator/EvalSpec#args).

Reply:
Thanks, I fixed it.